lib/message_format/parser.rb

```diff
@@ -69,7 +70,7 @@ def skip_whitespace ()
     def parse_text ( parent_type )
       is_hash_special = (parent_type == 'plural' or parent_type == 'selectordinal')
       is_arg_style = (parent_type == 'style')
-      text = ''
+      text = +''
```

Comment:
@froydnj I'm not familiar with this syntax, what does it do?

Reply:
It [duplicates frozen strings](https://devdocs.io/ruby~3.1/string#method-i-2B-40); we need a fresh, non-frozen string that we can append to.